.github/workflows/build-test.yml

```diff
@@ -47,3 +47,13 @@ jobs:
 
     - name: Test
       run: cargo test --verbose
+
+  format:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Rust linter
+      # make this command fail if cargo fmt had to make changes
+      run: cargo fmt && git status --short | wc -l | awk '{if ($1 != 0) {exit 1;}}'
```

Comment:
Optional: Here's a way I think to do this check in pure git, without wc and awk:

```
  - git update-index --refresh
  - git diff-index --exit-code HEAD
```

The second line returns nonzero if there are changes.  An advantage of this approach is that you should be able to see the diff in the CI output.

Reference: https://unix.stackexchange.com/a/394674/22110'

Please double-check that this works.

Reply:
Done.

Both lines taken separately produce the intended behavior of a more robust check.  Both commands repeat the affected file names to the console, so that would show up in CI.  I went with the second line, as suggested, even though it was slightly more verbose, b/c it also included the commit hash.